sgt-back-end/index.js

```diff
@@ -0,0 +1,148 @@
+const pg = require('pg');
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const express = require('express');
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = req.body.score;
+  if (!name || !course || !score || Number(score) < 0 ||
+  Number(score) > 100) {
+    res.status(400).json({
+      error: 'name, course, and valid score(0-100) must be provided.'
+    });
+    return;
+  }
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values ('${name}', '${course}', '${score}')
```

Comment:
It's dangerous to put values directly into your SQL because it opens your app up to SQL injection attacks. Instead, use a params array and placeholders

```sql
const sql = `
  insert into "grades" ("name", "course", "score")
  values ($1, $2, $3)
  returning *
`;

const params = [name, course, score];
```

Reply:
ohhh okay I see yea I should've know that rule applied here too